src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -1108,45 +1169,48 @@ void func_80A772EC(EnIk* this, GlobalContext* globalCtx) {
     Audio_PlaySoundGeneral(NA_SE_EN_IRONNACK_DEAD, &D_80A78FA0, 4, &D_801333E0, &D_801333E0, &D_801333E8);
 }
 
-void func_80A7735C(EnIk* this, GlobalContext* globalCtx) {
+// Cutscene: starts after final hit to Nabooru
+void EnIk_CsAdvanceTo05(EnIk* this, GlobalContext* globalCtx) {
     s32 pad[3];
-    f32 frames = Animation_GetLastFrame(&object_ik_Anim_0203D8);
+    f32 frames = Animation_GetLastFrame(&gIronKuckleNabooruDeathAnim);
 
     SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_ik_Skel_0205C0, NULL, this->jointTable, this->morphTable,
                        30);
-    Animation_Change(&this->skelAnime, &object_ik_Anim_0203D8, 1.0f, 0.0f, frames, ANIMMODE_ONCE, 0.0f);
-    this->action = 3;
+    Animation_Change(&this->skelAnime, &gIronKuckleNabooruDeathAnim, 1.0f, 0.0f, frames, ANIMMODE_ONCE, 0.0f);
+    this->action = IK_ACTION_3;
     this->drawMode = 2;
-    func_80A770C0(this, globalCtx, 4);
-    func_80A772EC(this, globalCtx);
+    EnIk_MoveNpcToPos(this, globalCtx, 4);
+    EnIk_CsPlaySfx_NabooruKnucleDefeat(this, globalCtx);
     this->actor.shape.shadowAlpha = 0xFF;
 }
 
-void func_80A77434(EnIk* this, GlobalContext* globalCtx) {
-    this->action = 4;
+// Cutscene: Armor falling off revealing Nabooru underneath
+void EnIk_CsAdvanceTo06(EnIk* this, GlobalContext* globalCtx) {
+    this->action = IK_ACTION_4;
     this->drawMode = 2;
-    func_80A772A4(this);
+    EnIk_CsPlaySfx_ArmorFallingOff(this);
     this->actor.shape.shadowAlpha = 0xFF;
 }
 
-void func_80A77474(EnIk* this, GlobalContext* globalCtx) {
-    this->action = 5;
-    this->drawMode = 0;
+// Cutscene: all the armor has fallen off
+void EnIk_CsAdvanceTo07(EnIk* this, GlobalContext* globalCtx) {
+    this->action = IK_ACTION_5;
+    this->drawMode = IK_DRAW_NOTHING;
     this->actor.shape.shadowAlpha = 0;
 }
 
-void func_80A7748C(EnIk* this, GlobalContext* globalCtx) {
-    func_80A77034(this, globalCtx);
-    func_80A779DC(this, globalCtx);
+void EnIk_UpdateCs_03(EnIk* this, GlobalContext* globalCtx) {
+    EnIk_UpdateBgCheckInfo(this, globalCtx);
+    EnIk_HandleSubscenesByNpcAction(this, globalCtx);
 }
 
-void func_80A774BC(EnIk* this, GlobalContext* globalCtx) {
-    func_80A7707C(this);
-    func_80A77034(this, globalCtx);
-    func_80A779DC(this, globalCtx);
+void EnIk_UpdateCs_04(EnIk* this, GlobalContext* globalCtx) {
+    EnIk_CheckSkelAnim(this);
+    EnIk_UpdateBgCheckInfo(this, globalCtx);
+    EnIk_HandleSubscenesByNpcAction(this, globalCtx);
 }
 
-void func_80A774F8(EnIk* this, GlobalContext* globalCtx) {
+void EnIk_UpdateCs_05(EnIk* this, GlobalContext* globalCtx) {
```

Comment:
what do these numbers in the cutscene related functions refer to?

Reply:
There are 6 different transitions in the various Nabooru fight, the first 3 are with Kotake and Komue, the last 3 are when the fight is over.

Reply:
I since realized these numbers are the corresponding indices in `sActionFuncs`

At first I thought they were related to the npcAction index which didn't make sense, but ok

Reply:
Actually no I'm confused again, for example `EnIk_CsAdvanceTo01` leads to `this->action = IK_ACTION_INIT;` (0) i.e. `EnIk_UpdateCs_00`

Reply:
bump

Reply:
Confused at what you are suggesting here, are you wanting the action enum to correspond to the function name?